addons/point_of_sale/models/pos_session.py

```diff
@@ -463,21 +456,26 @@ def _post_statement_difference(self, amount, is_opening):
                         _('Please go on the %s journal and define a Loss Account. This account will be used to record cash difference.',
                           self.cash_journal_id.name))
 
-                st_line_vals['payment_ref'] = _("Cash difference observed during the counting (Loss)") + (_(' - opening') if is_opening else _(' - closing'))
-                if not is_opening:
-                    st_line_vals['counterpart_account_id'] = self.cash_journal_id.loss_account_id.id
+                st_line_vals['payment_ref'] = _("Cash difference observed during the counting (Loss) - closing")
+                st_line_vals['counterpart_account_id'] = self.cash_journal_id.loss_account_id.id
             else:
                 # self.cash_register_difference  > 0.0
                 if not self.cash_journal_id.profit_account_id:
                     raise UserError(
                         _('Please go on the %s journal and define a Profit Account. This account will be used to record cash difference.',
                           self.cash_journal_id.name))
 
-                st_line_vals['payment_ref'] = _("Cash difference observed during the counting (Profit)") + (_(' - opening') if is_opening else _(' - closing'))
-                if not is_opening:
-                    st_line_vals['counterpart_account_id'] = self.cash_journal_id.profit_account_id.id
+                st_line_vals['payment_ref'] = _("Cash difference observed during the counting (Profit) - closing")
+                st_line_vals['counterpart_account_id'] = self.cash_journal_id.profit_account_id.id
+
+            created_line = self.env['account.bank.statement.line'].create(st_line_vals)
 
-            self.env['account.bank.statement.line'].create(st_line_vals)
+            if created_line:
+                statement = self.env['account.move'].search([('statement_line_id', '=', created_line.id)])
+                statement.message_post(body=_(
+                    "Related Session: %(link)s",
+                    link=self._get_html_link()
+                ))
```

Comment:
```suggestion
                created_line.move_id.message_post(body=_(
                    "Related Session: %(link)s",
                    link=self._get_html_link()
                ))
```

No need to search for the linked account.move, you can access it from the statement.line's move_id field.